### PR TITLE
Faisal/upgrade handler

### PR DESCRIPTION
**Pull request by faisalnaveedRNS**
## Description

Added upgrade handler doc

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by MikeSofaer — APPROVED** [review]

File: app/upgrades/v4/upgrades.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+		amt := sdk.NewCoins(sdk.NewCoin(types.PylonsCoinDenom, amountUpylons))
+		err := pylons.MintCoinsToAddr(ctx, addr, amt)
+		if err != nil {
+			return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+		}
+	}
+	return nil
+}
+
+// Mint uplyons for address with valid (with IAP))
```
Comment (afti-githobo) [inline app/upgrades/v4/upgrades.go:278]: upylons

File: app/upgrades/v4/upgrades_test.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+				addr, _ := sdk.AccAddressFromBech32(item.Creator)
+				accAmount := suite.App.BankKeeper.GetBalance(suite.Ctx, addr, types.PylonsCoinDenom)
+				suite.Require().Equal(accAmount.Amount, sdk.NewInt(int64(amountValid)))
+			}
+		})
+	}
+}
+
+// set up googleIAPOder
+func (suite *UpgradeTestSuite) setUpGoogleIAPOder(ctx sdk.Context, n int, productID string) []types.GoogleInAppPurchaseOrder {
```
Comment (afti-githobo) [inline app/upgrades/v4/upgrades_test.go:195]: setUpGoogleIAPOrder

File: app/upgrades/v4/upgrades.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+	return math.ZeroInt()
+}
+
+func RefundLuxFloralis(ctx sdk.Context, pylons *pylonskeeper.Keeper) {
+	// Get all execute recipe history by cookbookid and recipe id
+	histories := pylons.GetAllExecuteRecipeHis(ctx, LuxFloralisCookBookID, LuxFloralisRecipeID)
+
+	// Looping execute recipe history to get sender address
+	for _, history := range histories {
+		amt := sdk.NewCoins(sdk.NewCoin(types.PylonsCoinDenom, sdk.NewInt(30000000)))
```
Comment (faisalnaveedRNS) [inline app/upgrades/v4/upgrades.go:316]: why the amount is hardcoded

File: app/upgrades/v4/upgrades.go [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+		TkPreSale,
+		TkCFSale,
+		TkRegDSSale,
+		CompanyRevenue,
+		EngineHotWal,
+	}
+
+	TotalUbedrock = math.NewIntFromUint64(1e+15) // 1 bedrock = 1_000_000 ubedrock
+
+	// UbedrockDistribute = map[string]math.Int{
```
Comment (kjawadDeveloper2) [inline app/upgrades/v4/upgrades.go:55]: Deadcode.

File: app/upgrades/v4/upgrades.go [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	bankKeeper bankkeeper.Keeper,
+	accKeeper *authkeeper.AccountKeeper,
+	staking *stakingkeeper.Keeper,
+	pylons *pylonskeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		// logger := ctx.Logger()
```
Comment (kjawadDeveloper2) [inline app/upgrades/v4/upgrades.go:93]: Dead code.

File: app/upgrades/v4/upgrades_test.go [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+			for _, item := range items {
+				addr, _ := sdk.AccAddressFromBech32(item.Creator)
+				accAmount := suite.App.BankKeeper.GetBalance(suite.Ctx, addr, types.PylonsCoinDenom)
+				suite.Require().Equal(accAmount.Amount, sdk.NewInt(int64(amountValid)))
+			}
+		})
+	}
+}
+
+// set up googleIAPOder
```
Comment (kjawadDeveloper2) [inline app/upgrades/v4/upgrades_test.go:189]: Typo googleIAPOrder.

**Review by kjawadDeveloper2 — APPROVED** [review]

**Pull request merged**